src/core/lib.lua

```diff
@@ -686,6 +686,38 @@ function getenv (name)
    else return nil end
 end
 
+function fstruct(def)
```

Comment:
At least add comments that describe usage and purpose of these functions, as of now I have no idea what they even do. And functions that are not `local` to `core.lib` indeed require documentation in `src/README.md.src`. Without the documentation I can hardly tell where the functions belong.

@andywingo Undocumented public functions in `core.lib` are a bug, I'd appreciate any tips on which functions are inadequately documented.